web/src/main/webapp/xslt/services/metadata/validate-fn.xsl

```diff
@@ -186,14 +187,10 @@
                   </xsl:if>
                 </xsl:when>
                 <xsl:when test="$errorType = 'enumeration-valid'">
-                  <xsl:value-of select="$strings/enum1"/> '<xsl:value-of
-                  select="regex-group(1)"/>'
-                  <xsl:value-of select="$strings/enum2"/>
-                  <xsl:value-of
-                    select="geonet:getTitleWithoutContext($schema, concat(regex-group(3), ':', regex-group(4)), $labels)"
-                  /> (<xsl:value-of select="concat(regex-group(3), ':', regex-group(4))"/>).
+                  <!--The topic category must be found in the following list-->
                   <xsl:value-of select="$strings/enum3"/>
-                  <xsl:value-of select="regex-group(2)"/>.
+                  <!--The code list with translation text-->
+                  (<xsl:value-of select="$codelists//codelist[@name='gmd:MD_TopicCategoryCode']/entry/value" separator=", " />)
```

Comment:
I don't believe you can hard code the reference the gmd:MD_TopicCategoryCode in the enumeration-valid error as the enumeration-valid is used to generate error messages for all kinds of invalid code list.  I believe this has to be a generic error handler for all enumeration-valid.
This  will most likely break validations of other code lists.

Reply:
The old code used <xsl:value-of select="regex-group(2)"/> as the list to parse. When I test it with <xsl:message> console output, none of this regex-group has got the proper value.